test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -1471,9 +1479,80 @@ public void testGetDocContentSiteCollection() throws Exception {
+            new GroupPrincipal("BUILTIN\\users", DEFAULT_NAMESPACE),
+            new UserPrincipal("GDC-PSL\\spuser4", DEFAULT_NAMESPACE)));
+      tmp.put(SITES_SITECOLLECTION_VISITORS, Arrays.<Principal>asList());
+      goldenGroups1 = Collections.unmodifiableMap(tmp);
+    }
+
+    final Map<GroupPrincipal, Collection<Principal>> goldenGroups2;
+    {
+      Map<GroupPrincipal, Collection<Principal>> tmp
+          = new TreeMap<GroupPrincipal, Collection<Principal>>();
+      tmp.put(SITES_SITECOLLECTION_OWNERS, Arrays.<Principal>asList(
+          GDC_PSL_ADMINISTRATOR));
+      tmp.put(SITES_SITECOLLECTION_MEMBERS, Arrays.asList(
+            new GroupPrincipal("BUILTIN\\users", DEFAULT_NAMESPACE),
+            new UserPrincipal("GDC-PSL\\spuser4", DEFAULT_NAMESPACE)));
+      tmp.put(SITES_SITECOLLECTION_REVIEWERS, Arrays.asList(
+            new UserPrincipal("GDC-PSL\\spuser2", DEFAULT_NAMESPACE),
+            new GroupPrincipal("BUILTIN\\users", DEFAULT_NAMESPACE)));
+      goldenGroups2 = Collections.unmodifiableMap(tmp);
+    }
+
+    ContentExchange scContentExchange1 = new ContentExchange(
+        ObjectType.SITE_COLLECTION, null, null, null, true, false, null,
+        loadTestString("sites-SiteCollection-sc.xml"), true);
+    ContentExchange scContentExchange2 = new ContentExchange(
+        ObjectType.SITE_COLLECTION, null, null, null, true, false, null,
+        loadTestString("sites-SiteCollection-sc-useonce.xml"), false);
+
+    SoapFactory siteDataFactory = MockSoapFactory.blank()
+        .endpoint(VS_ENDPOINT, MockSiteData.blank()
+            .register(VS_CONTENT_EXCHANGE)
+            .register(CD_CONTENT_EXCHANGE)
+            .register(ROOT_SITE_SAW_EXCHANGE)
+            .register(SITES_SITECOLLECTION_SAW_EXCHANGE))
+        .endpoint(SITES_SITECOLLECTION_ENDPOINT, MockSiteData.blank()
+            .register(SITES_SITECOLLECTION_URLSEG_EXCHANGE)
+            .register(SITES_SITECOLLECTION_S_CONTENT_EXCHANGE)
+            .register(scContentExchange1)
+            .register(scContentExchange2));
+
+    adaptor = new SharePointAdaptor(siteDataFactory,
+        new UnsupportedHttpClient(), executorFactory,
+        new MockAuthenticationClientFactoryForms(),
+        new UnsupportedActiveDirectoryClientFactory());
+
+    RecordingDocIdPusher pusher = new RecordingDocIdPusher();
+    adaptor.init(new MockAdaptorContext(config, pusher));
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    GetContentsRequest request = new GetContentsRequest(
+        new DocId("http://localhost:1/sites/SiteCollection"));
+    GetContentsResponse response = new GetContentsResponse(baos);
+    // First call returns 3 groups(Owners, Members with 2 users and Visitors)
+    adaptor.getDocContent(request, response);
+    assertEquals(goldenGroups1, pusher.getGroupDefinitions());
+
+    // Second call returns 3 groups(Owners, Members with 1 user and
+    // Reviewers with 1 user)
+    adaptor.getDocContent(request, response);
```

Comment:
You should get a new GetContentsResponse here, they're not really meant to be reusable. We're just not looking at the recorded data here. Plus, RecordingResponse is truly not reusable, so we would have to change when we switch to that. The other tests here get a new one if they call getDocContent again.

Reply:
Thanks. Done.